### DISCUSSION TITLE
How to update project made with previous version of golem when newer version of golem is released

### DISCUSSION THREAD

**Discussion by isaactpetersen** [ThinkR-open/golem#667] [Q&A]
Thanks for your development of this very helpful package. I'm relatively new to the `golem` package, so I apologize for the novice question but I haven't seen it discussed in the documentation materials. I developed a Shiny app using `golem` version 0.2.1. A new version was just released (0.3.1), which I look forward to using. I know how to update the `golem` package, but how do I update the Shiny app I created with the previous version of golem to be based on the newer version of `golem'? Is there a straightforward way, or should I create an entirely new project and go through all of the steps again? Thanks in advance!

**Comment by ColinFay** [comment]
Thanks @isaactpetersen! 

That's indeed not discussed because there is no breaking changes in `{golem}` that would break your application (at least that I'm aware of) when it comes to the deployment tools. 

Here is what I suggest: 

- update `{golem}`
- test the app and see if it works

If ever it doesn't work, 
- switch back to `remotes::install_version("golem", version = "0.2.1")`
- come back here, and we'll debug together what might be broken

Doest that sound good to you?

Cheers,

Colin

**Comment by isaactpetersen** [comment]
Hi Colin,

That all sounds good. I can confirm that (so far) the Shiny app works fine when I update `golem`. In any case, the foundation of the R scripts in the app were generated using the prior version of the `golem` package. If I wanted to update the scripts in the app to be generated from the new version of the `golem` app, would I need to generate a new script for each of the files that had been generated using the old version of the package? For instance, would I need to re-run `golem::add_module(name = nameOfModule)` using the new version of `golem` to create an updated version of the module script so that it is based on the latest version of the golem package? I imagine there would be benefits to regenerating the scripts to be used in my app using the newest version of `golem` to allow using the latest functionality etc. I have no idea whether the functionality changed with `golem::add_module()` in the latest `golem` update, but that is just an example. Basically, how can I update my app code to be based on the latest version of the `golem` package after I've updated `golem` to the latest version? Does that make sense what i'm asking? I apologize--I'm struggling to articulate my question. Thanks very much for your help!

**Comment by ShinyFabio** [comment]
I'm also interested

**Comment by ColinFay** [comment]
hey @isaactpetersen and @ShinyFabio, 

There should be no reason to update your old code — what the new version of `{golem}` does, for module, is switching to the new module skeleton, but that doesn't prevent from keeping the old structure, and you can mix both (the new moduleServer and the old callModule syntaxes) in your app: so even if you chose to create new modules, you can keep the old ones as they are. 

The only change is (and it comes from `{shiny}`, not `{golem}`) is with `options(shiny.autoload.r)` that might conflict if you are deploying on an RStudio platform. 
In that case, you might want to rerun the `golem::add_rstudioconnect()` (or shinyapps or shinyserver) function, and/or run `golem::disable_autoload()`.

For the rest, `{golem}` has been updated for the __building__ of the application, not for the __deployment__, meaning that working code should still be working, with no need for update :) 

Let me know if that answers your q, 

Colin
